src/nccl_ofi_net.c

```diff
@@ -1063,6 +1065,8 @@ static ncclResult_t ofi_init(ncclDebugLogger_t logFunction)
 		cuda_flush = true;
 #endif
 	}
+	
+	gdr_flush = !ofi_nccl_gdr_flush_disable();
```

Comment:
nit: Please add a comment that this is a user configuration to disable flush operations and a warning to indicate that this might lead to data inconsistency.

Reply:
@rashikakheria - is there a known app that will have a corruption problem if the flush is disabled? We're trying to determine (a) whether the problem exists with PSM3 and (b) how to respond if it does. I'd also like to inquire about under what circumstances cudaDeviceFlushGPUDirectRDMAWrites() causes hangs.

Reply:
The nccl tests with data checking enabled should be sufficient to see the issue.  You don't need an application (and I suspect the benchmarks will make it even more likely to happen, but could be wrong).

Reply:
> The nccl tests with data checking enabled should be sufficient to see the issue. You don't need an application (and I suspect the benchmarks will make it even more likely to happen, but could be wrong).

So, that's the thing. The tests all pass in our environment. We just noticed that on one particular machine, PSM3 was issuing debug messages noting that it wasn't actually doing the 0-length reads because the MR is a NULL pointer. That led to me creating this pull request and started us down the conversation of whether or not this is necessary or not.

Reply:
Is PSM3 using the simple protocol at any point (I assume yes, or you wouldn't have seen the flush at all).  This is probably a better question for @jdinan, as the race isn't with Libfabric or PCIe or NICs, but inside the GPU itself (well, and behaviors of NCCL).

Reply:
Anyway, thanks for the suggestion. I'll grab the one "odd" machine, do a new build of NCCL, and run the tests again, making sure data-checking is enabled.

Reply:
If the provider doesn't set `FI_MR_HMEM` mode bit for local reads then they should be registering it. Isn't that the right expectation from apps?